### PR TITLE
Feature/git persist extended

### PR DESCRIPTION
**Pull request by ftdholbrook**
- add support to persist TenantProperties/Tenant
- change API to accept full TenantConfiguration
- still TODO localization

### CODE REVIEW

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/model/ApplicationTenantConfigurationPersistence.java [hunk 1]
```diff
@@ -59,4 +64,12 @@ public void setDatasources(Map<String, Map<String, Map<String, DataSourceElement
     public void setArchive(Map<String, Map<String, ArchiveProperties>> archive) {
         this.archive = archive;
     }
+
+    public Map<String, Map<String, Tenant>> getTenantProperties() {
```
Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/model/ApplicationTenantConfigurationPersistence.java]: perhaps javadocs so i can understand what the return is, exactly? Is it map of tenantKey to the map of tenantKey to tenant? Why would each tenant have a different set of tenant maps??

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/git/TenantConfigurationPersistenceServiceGit.java [hunk 2]
```diff
@@ -30,6 +34,7 @@
 
     private static final Logger logger = LoggerFactory.getLogger(TenantConfigurationPersistenceServiceGit.class);
     private static final String configFileName = "application.yml";
+    private static final String tenantsFileName = "application.yml";
```
Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/git/TenantConfigurationPersistenceServiceGit.java]: I have a completely indefensible nit: can these be `ConfigFileName` and `TenantsFileName`? I really dislike the ALL_CAPS const thing but, for whatever reason, i like the leading uppercase.

File: admin-service/src/test/java/org/opentestsystem/rdw/admin/multitenant/git/TenantConfigurationPersistenceServiceGitIT.java [hunk 3]
```diff
@@ -104,10 +106,17 @@ public void testSaveFile() throws Exception {
         applicationTenantConfiguration.setDataSources(ImmutableMap.of("reporting_ro", dataSourceElements));
         applicationTenantConfiguration.setArchiveProperties(archivePropertiesTenant);
 
-        ApplicationTenantConfigurationPersistence applicationTenantConfigurationPersistence = new ApplicationTenantConfigurationPersistence("CAX", applicationTenantConfiguration);
+        TenantProperties.Tenant tenant = new TenantProperties.Tenant();
+        tenant.setId("CAX");
+        tenant.setKey("CAX");
+        tenant.setName("California Test");
```
Comment (malaffoon) [inline admin-service/src/test/java/org/opentestsystem/rdw/admin/multitenant/git/TenantConfigurationPersistenceServiceGitIT.java]: I ran into this too, perhaps we can add a constructor that takes the three values id, key, name?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/git/TenantConfigurationPersistenceServiceGit.java (hunk 2 above)
Comment (ftdholbrook) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/git/TenantConfigurationPersistenceServiceGit.java]: really?  I think that is the worst of all worlds, but I'll do it.

Comment (burzum619) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/git/TenantConfigurationPersistenceServiceGit.java]: What's wrong with the `ALL_CAPS_CONSTS`?   🤔

Comment (ftdholbrook) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/git/TenantConfigurationPersistenceServiceGit.java]: These are private string literals, there's nothing special about them, they aren't part of a public api.  Coming from an immutable by default language (where you signal the exceptional mutability) it strikes me as really weird treating stuff like this as special. :-)

Comment (malaffoon) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/git/TenantConfigurationPersistenceServiceGit.java]: Like i said, it's indefensible. I'm fine with leaving them.

**Pull request merged**